### PR TITLE
fix(m1): correctness fixes across query, kit, expr, schema, and driver

### PR DESCRIPTION
**Pull request by charles-fineman**
## Summary

M1 milestone — all correctness fixes that should land before any new feature work.

### query
- **#3** `structSetsForUpdate` returns error on nil/non-struct/invalid input instead of panicking
- **#119** Empty `DO UPDATE SET` from invalid input returns error rather than emitting broken SQL
- **#120** Added `!rv.IsValid()` guard test
- **#11** Set-operation `ORDER BY` strips table qualifier via `ToSQLUnqualified()` — fixes invalid SQL for UNION/INTERSECT/EXCEPT
- **#134** Edge-case tests for unqualified ORDER BY
- **#128** `OnConflict` typed column API is canonical; string-based variant reconciled

### kit
- **#6** `constraintMap` uses a synthetic `<type>:<name>` key — prevents collisions between different constraint kinds with the same name
- **#8** `Diff()` sorts output slices for deterministic ordering
- **#9** `constraintsEqual` now compares FK `OnDelete`/`OnUpdate` fields

### expr
- **#93** `Lead`/`LagWithDefault` bind the default value through `ctx.Add()` — fixes SQL injection risk
- **#99** `NthValue` validates `n ≥ 1` at construction time
- **#104** Window frame sentinel values (`UnboundedPreceding` etc.) are now immutable accessor functions
- **#129** `RawArgs` panics at query-construction time if placeholder count ≠ arg count
- **#131** `AliasedCol` only emits `AS alias` in SELECT position; bare column reference elsewhere
- **#101 / #106** Structural DDL ordering improved (CREATE TABLE before ALTER TABLE/constraints). Full fix for view ordering deferred to M3 — requires `Views` in `Snapshot` (tracked in #39)

### schema/pg
- **#13** Removed `CheckBuilder` and `setReferences` dead stubs
- **#111** `EnumColumnBuilder.Default` escapes single quotes in enum values
- **#118** Empty enum value list returns an error

### driver/pgx
- **#12** `PreparedSelect`/`PreparedExec` use `p.sql` (the SQL string) instead of `p.name` for execution — fixes cross-connection failures when prepared statements aren't cached on every connection

## Test plan

- [ ] `go test ./...` passes (all tests green on branch)
- [ ] Review `query/update_test.go` for new nil/invalid/non-struct cases (#3/#119/#120)
- [ ] Review `query/setop_test.go` for unqualified ORDER BY cases (#11/#134)
- [ ] Review `expr/window_test.go` for Lead/Lag default binding, NthValue, AliasedCol
- [ ] Review `kit/diff_test.go` for deterministic output and FK equality
- [ ] Spot-check `schema/pg/column.go` for removed stubs and enum escaping

### CODE REVIEW

File: driver/pgx/prepared.go [hunk 1]
```diff
@@ -64,21 +64,27 @@ func (p *PreparedSelect[T]) Name() string { return p.name }
 func (p *PreparedSelect[T]) SQL() string { return p.sql }
 
 // QueryAll executes the prepared query and returns all matching rows.
+//
+// The SQL string is used for execution rather than the statement name.
+// pgxpool connections are per-connection; named prepared statements are
+// not reliable across connections in the pool. Using the SQL directly
+// lets pgx v5's per-connection statement cache handle re-preparation
+// transparently on new pool connections (Fix #12).
```
[the remark this answers is not in the document]
Comment (scottescue) [reply driver/pgx/prepared.go]: @charles-fineman-bot Is this change consistent with the spec and in parity with Drizzle?
